### PR TITLE
refactor: CCB warnings and lint fixes

### PR DESCRIPTION
**Pull request by jackgerrits**

### CODE REVIEW

File: vowpalwabbit/conditional_contextual_bandit.cc [hunk 1]
```diff
@@ -36,11 +36,11 @@ using namespace VW::config;
 namespace logger = VW::io::logger;
 
 template <typename T>
-void return_v_array(v_array<T>&& array, VW::v_array_pool<T>& pool)
+void return_v_array(v_array<T>& array, VW::v_array_pool<T>& pool)
 {
   array.clear();
   pool.reclaim_object(std::move(array));
-  array.clear();
+  array = v_array<T>{};
```
Comment (jackgerrits) [inline vowpalwabbit/conditional_contextual_bandit.cc:43]: This fixes the use after move which although may be valid in this case, most linters are going to warn loudly for it

**Review by lalo — APPROVED** [review]

File: vowpalwabbit/conditional_contextual_bandit.cc [hunk 2]
```diff
@@ -130,7 +130,7 @@ bool split_multi_example_and_stash_labels(const multi_ex& examples, ccb& data)
       case example_type::slot:
         data.slots.push_back(ex);
         break;
-      default:
+      case unset:
```
Comment (lalo) [inline vowpalwabbit/conditional_contextual_bandit.cc:133]: what is unset

Comment (jackgerrits) [reply vowpalwabbit/conditional_contextual_bandit.cc:133]: It is the other enum value

Comment (lalo) [reply vowpalwabbit/conditional_contextual_bandit.cc:133]: we dont need a default case anymore?

Comment (jackgerrits) [reply vowpalwabbit/conditional_contextual_bandit.cc:133]: Yeah now that I look at this again it seems perhaps the `default` was better and less likely to be misused in future

**Review by lalo — APPROVED** [review]

**Pull request merged**